### PR TITLE
adpd410x support

### PR DESCRIPTION
**Pull request by amiclaus**
# Description

Add pyadi-iio support for the adpd410x multimodal sensor front end.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How has this been tested?

Tested in conjunction with https://github.com/analogdevicesinc/EVAL-ADICUP3029/tree/master/projects/ADuCM3029_demo_cn0503

and https://github.com/analogdevicesinc/pyadi-iio/blob/adpd410x/examples/adpd410x_example.py

**Test Configuration:**

* Hardware:  CN0503 + EVAL-ADICUP3029
* OS: Windows

# Documentation

https://wiki.analog.com/resources/tools-software/uc-drivers/adpd410x

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: adi/adpd410x.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    @last_timeslot.setter
+    def last_timeslot(self, value):
+        """Sets last timeslot of the adpd410x"""
+        self._set_iio_dev_attr_str("last_timeslot", value, self._rxadc)
+
+    @property
+    def operation_mode(self):
+        """Get operation mode of the adpd410x"""
+        return None
```
Comment (tfcollins) [inline adi/adpd410x.py:88]: Can this be read back?

Comment (amiclaus) [reply adi/adpd410x.py:88]: Currently the driver is populated with setters only:
https://github.com/analogdevicesinc/no-OS/blob/5037d892196ec5c692452a624cc11d1b2561f724/drivers/photo-electronic/adpd410x/adpd410x.h#L1214-L1223

I believed the implementation of getters/adding device parameters should start from the no-OS driver and then added into the upper layers (iio no-OS driver, pyadi-iio support).

If you think it is necessary to have the readbacks, I can do the updates in the files mentioned above.

Comment (tfcollins) [reply adi/adpd410x.py:88]: Does that mean the reads will fail or they are meaningless?

Comment (amiclaus) [reply adi/adpd410x.py:88]: They are meaningless at this point in time. (Just added them as placeholders for future implementation).
I made some updates on the no-OS driver so I can add the read backs here also. Will update soon.

**Review by amiclaus — COMMENTED** [review]
V2:
- populate readbacks accordingly

**Review by amiclaus — COMMENTED** [review]
V3:
- solve newly appeard conflicts

**Pull request merged**